app/lib/bk/compat/parsers/circleci/orbs.rb

```diff
@@ -28,6 +19,144 @@ def translator(action, _config)
           ]
         end
       end
+
+      # Handles Docker installation logic
+      class DockerInstaller
+        DOCKER_INSTALL_PARAMS = {
+          'install-dir' => {
+            'type' => 'string',
+            'description' => 'Directory in which to install Docker binaries',
+            'default' => '/usr/local/bin'
+          },
+          'version' => {
+            'type' => 'string',
+            'description' => 'Version of Docker to install',
+            'default' => 'latest'
+          }
+        }.freeze
+
+        def initialize(config = {})
+          @config = config
+        end
+
+        def custom_directory?
+          return false unless @config.key?('install-dir')
+
+          @config['install-dir'] != DOCKER_INSTALL_PARAMS['install-dir']['default']
+        end
+
+        def specific_version?
+          return false unless @config.key?('version')
+
+          @config['version'] != DOCKER_INSTALL_PARAMS['version']['default']
+        end
+
+        def install_commands
+          return simple_install unless custom_directory? || specific_version?
+
+          generate_advanced_install
+        end
+
+        private
+
+        def simple_install
+          [
+            "echo '~~~ Installing Docker'",
+            'curl -fsSL https://get.docker.com -o get-docker.sh',
+            'sh get-docker.sh'
+          ]
+        end
+
+        def generate_advanced_install
+          [
+            generate_script_header,
+            generate_custom_dir_commands,
+            generate_version_specific_commands
+          ].flatten.compact
+        end
+
+        def generate_script_header
+          [
+            '#!/usr/bin/env bash',
```

Comment:
this project does not generate scripts so the shebang is definitely not necessary